runtime/src/deterministic.rs

```diff
@@ -46,6 +48,49 @@ use std::{
 };
 use tracing::trace;
 
+
+#[derive(Clone, Debug, Hash, PartialEq, Eq)]
+struct BandwidthLabels {
+    socket: SocketAddr,
+    direction: String,
+}
+
+impl EncodeLabelSet for BandwidthLabels {
+    fn encode(&self, mut encoder: prometheus_client::encoding::LabelSetEncoder) -> Result<(), std::fmt::Error> {
+        encoder.encode_label();
+        encoder.encode_label();
+        Ok(())
+    }
+}
+
+#[derive(Clone, Debug)]
```

Comment:
I would remove the `Clone` derivation here. I think the `Arc<Metrics>` below should be sufficient to provide necessary access over clones of `Context`.